pages/experiments.js

```diff
@@ -1,22 +1,67 @@
 import Head from "next/head";
+import { useState } from "react";
 import { serverSideTranslations } from "next-i18next/serverSideTranslations";
 import { useTranslation } from "next-i18next";
 import { Layout } from "../components/organisms/Layout";
-import { Experiment } from "../components/atoms/Experiment";
+import { Experiment } from "../components/molecules/Experiment";
 import { useRouter } from "next/router";
+import { Filter } from "../components/molecules/Filter";
 
 export default function experiments(props) {
   const { t } = useTranslation("common");
   const { asPath } = useRouter();
+  const [filter, setFilter] = useState("all");
+  const [filteredExperiments, setFilteredExperiments] = useState(
+    props.experimentData
+  );
+
+  // get the filters from the data
+  const filters = Object.values(
+    props.experimentData.reduce(
+      (filters, { tag }) => {
+        if (!filters[tag]) {
+          filters[tag] = {
+            id: tag,
+            label: t(tag),
+            checked: filter === tag,
+          };
+        }
+        return filters;
+      },
+      {
+        all: {
+          id: "all",
+          label: "All",
+          checked: filter === "all",
+        },
+      }
+    )
+  );
```

Comment:
okay, so just humour me. 

It looks like whenever you click a particular filter, this loop re-executes every time, always coming up with the same result (eg, the same set of 4 filters). (ie, if you console.log on line 22 you see it). Should that be updating each time a filter is clicked? I get you redraw the experiments underneath, but is the point here that we want to set "checked" on whichever filter is tagged this way?

Seems like it's fine either way, was just curious.

Reply:
So we want to be able to add filters dynamically based on the content from the CMS this is the way to do that. Considering we're only going to have 3-4 elements maximum. This is almost a constant time approach